python/ray/serve/_private/router.py

```diff
@@ -689,17 +691,142 @@ def _get_singleton_asyncio_loop(cls) -> asyncio.AbstractEventLoop:
     def running_replicas_populated(self) -> bool:
         return self._asyncio_router.running_replicas_populated()
 
+    def _run_coroutine_threadsafe(self, coro, loop, asyncio_future_callback=None):
```

Comment:
This util has nothing to do with the router itself. Let's extract it to common utils

Reply:
Actually, we don't even need to override this method:

1. Wrap the passed coro into Task yourself
2. Attach the callback
3. Pass it into vanilla _run_coroutine_threadsafe

Reply:
Moved `_run_coroutine_threadsafe` to utils, and based on @alexeykudinkin 's offline suggestion, I removed the callback logic from `_run_coroutine_threadsafe`, keeping to the minimal to include support for `asyncio futures` as input